server/vulnerabilities/cpe.go

```diff
@@ -170,7 +170,7 @@ func CPEFromSoftware(db *sqlx.DB, software *fleet.Software) (string, error) {
 	checkTargetSW := ""
 	args := []interface{}{onlyAlphaNumeric.ReplaceAllString(cleanAppName(software.Name), " ")}
 	if targetSW != "" {
-		checkTargetSW = " AND target_sw MATCH ?"
+		checkTargetSW = " AND target_sw = ?"
```

Comment:
What about for:
```go
	case "programs":
		targetSW = "windows*"
```
Also, `MATCH` doesn't work with `"node.js"`, do we need to escape it?

Reply:
> It doesn't work with it, but = does.

Mhm... does it?
```
sqlite> SELECT count(*) FROM cpe WHERE rowid in (SELECT rowid FROM cpe_search WHERE target_sw MATCH '"windows*"') order by deprecated asc;
4250

sqlite> SELECT count(*) FROM cpe WHERE rowid in (SELECT rowid FROM cpe_search WHERE target_sw = 'windows*') order by deprecated asc;
0

sqlite> SELECT count(*) FROM cpe WHERE rowid in (SELECT rowid FROM cpe_search WHERE target_sw MATCH '"test.js"') order by deprecated asc;
0

sqlite> SELECT count(*) FROM cpe WHERE rowid in (SELECT rowid FROM cpe_search WHERE target_sw MATCH '"node.js"') order by deprecated asc;
40154
```
Source: https://stackoverflow.com/questions/45681645/how-to-enable-fts5-search-a-string-with-and-0-9

So should stick to `MATCH` and quote the `targetSW` for the exact matches:
```go
	switch software.Source {
	case "apps":
		targetSW = "\"macos\""
        [...]
	case "npm_packages":
		targetSW = "\"node.js\""
	case "atom_packages":
	case "programs":
		targetSW = "\"windows\"*" // asterisk outside quotes
	case "ie_extensions":
	case "chocolatey_packages":
	}
```

Reply:
Darn... thank you for double checking me... "never trust the docs" is the lesson I'm taking away.

And also that we don't have good enough tests. Will update with this.